jujugui/static/gui/src/app/init/component-renderers-mixin.js

```diff
@@ -94,13 +94,12 @@ const ComponentRenderersMixin = superclass => class extends superclass {
         visible={db.services.size() > 0}>
         <AddedServicesList
           changeState={this._bound.changeState}
-          findRelatedServices={db.findRelatedServices.bind(db)}
-          findUnrelatedServices={db.findUnrelatedServices.bind(db)}
           hoveredId={hoveredId}
-          hoverService={ServiceModule.hoverService.bind(ServiceModule)}
-          panToService={ServiceModule.panToService.bind(ServiceModule)}
-          services={db.services}
-          updateUnitFlags={db.updateUnitFlags.bind(db)} />
+          serviceModule={shapeup.addReshape({
```

Comment:
Can't you use fromShape here?
`serviceModule={shapeup.fromShape(ServiceModule, AddedServicesList.propTypes.serviceModule)}` ?

Reply:
Yes! Thanks, fixed.